python-package/lightgbm/sklearn.py

```diff
@@ -597,7 +597,9 @@ def _process_params(self, stage: str) -> Dict[str, Any]:
         if callable(self._objective):
             if stage == "fit":
                 self._fobj = _ObjectiveFunctionWrapper(self._objective)
-            params['objective'] = 'None'  # objective = nullptr for unknown objective
+                params['objective'] = self._fobj
+            elif stage == "predict":
+                params['objective'] = 'None'
```

Comment:
```suggestion
            else:
                params['objective'] = 'None'
```

Instead of adding a new `if` branch here, is it possible to just indent the following line over so that it's only run when `stage == "fit"`? Since I don't think `objective` should ever need to be passed through `objective` when predicting.

https://github.com/microsoft/LightGBM/blob/17d4e007852b4f0b34b211a674cb26e6751c98e2/python-package/lightgbm/sklearn.py#L604

That would make this logic a bit simpler, by avoiding adding another `if` branch.

Reply:
ok, agree on it.

The idea is to avoid the `_ObjectiveFunctionWrapper` type to be passed as an `objective` parameter when calling `predict`. If you try to convert these type to string to be passed to C api it fails.

Reply:
> Since I don't think `objective` should ever need to be passed through `objective` when predicting.

Then I strongly prefer just `pop()` `objective` key because `'None'` has special meaning in LightGBM.